internal/metrics/url_response.go

```diff
@@ -0,0 +1,57 @@
+package metrics
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	externalURLs = []string{"https://httpstat.us/503", "https://httpstat.us/200"}
+
+	availability = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "reference_addon_sample_availability",
+			Help: "external url availability.",
```

Comment:
Can we add `status == 0` means down. It can get confusing as UNIX 0 exit code means all is good :)